adr/0002-resolve-schema-references-lazily.md

```diff
@@ -0,0 +1,19 @@
+# 3. Resolve schema references lazily
+
+Date: 2019-08-08
+
+## Status
+
+Accepted
+
+## Context
+
+We need
+
+## Decision
+
+We will
+
+## Consequences
+
+What becomes easier or more difficult to do and any risks introduced by the change that will need to be mitigated.
```

Comment:
Test collection phase becomes faster, also it allows to avoid unnecessary schema resolving if certain tests are de-selected.